src/clients/java/src/main/java/com/tigerbeetle/BlockingRequest.java

```diff
@@ -143,7 +143,7 @@ private void waitForCompletionUninterruptibly() {
         }
     }
 
-    TResponse getResult() throws RequestException {
+    TResponse getResult() {
 
         assertTrue(result != null || exception != null, "Unexpected request result: result=null");
```

Comment:
I am wondering... Do we actually need this exception as a separate class even? From that

>Check if this client is compatible with the server's version.

It feels like the original intention behind the exception was to signal error response from the server, and then it mutated to signal local programming errors. In this new role, perhaps we can just throw an `AssertionError` for non-ok packet statuses?

Reply:
The original intention was to map result codes into Java exceptions: `InitializationException` for `tb_client_init`, `RequestException` for `packet.status`, and `ConcurrencyExceededException` for `tb_client_acquire_packet`.

> Check if this client is compatible with the server's version.

This message is indeed wrong, we never planned to return error response from the server at the protocol level (e.g. invalid size or operation), I think the C# client uses the same string.

(...)

I completely agree with `RequestException extends RuntimeException`, but I don't have a strong opinion on using `AssertionError` for it (feels like we are losing information if we reduce the packet status to an assertion error message).